service/src/test/java/bio/terra/pipelines/stairway/imputation/steps/SubmitQuotaConsumedSubmissionStepTest.java

```diff
@@ -0,0 +1,242 @@
+  @Mock private RawlsService rawlsService;
+  @Captor private ArgumentCaptor<SubmissionRequest> submissionRequestCaptor;
+  @Captor private ArgumentCaptor<MethodConfiguration> updateMethodConfigCaptor;
+  @Captor private ArgumentCaptor<MethodConfiguration> setMethodConfigCaptor;
+  @Mock private SamService samService;
+  @Mock private FlightContext flightContext;
+
+  private final UUID testJobId = TestUtils.TEST_NEW_UUID;
+
+  @BeforeEach
+  void setup() {
+    FlightMap inputParameters = new FlightMap();
+    FlightMap workingMap = new FlightMap();
+
+    when(flightContext.getInputParameters()).thenReturn(inputParameters);
+    when(flightContext.getWorkingMap()).thenReturn(workingMap);
+    when(samService.getTeaspoonsServiceAccountToken()).thenReturn("thisToken");
+  }
+
+  @Test
+  void doStepSuccess() {
+    // setup
+    StairwayTestUtils.constructCreateJobInputs(flightContext.getInputParameters());
+    when(flightContext.getFlightId()).thenReturn(testJobId.toString());
+    MethodConfiguration returnedMethodConfiguration = new MethodConfiguration();
+    when(rawlsService.getCurrentMethodConfigForMethod(
+            "thisToken",
+            TestUtils.CONTROL_WORKSPACE_BILLING_PROJECT,
+            TestUtils.CONTROL_WORKSPACE_NAME,
+            SubmitQuotaConsumedSubmissionStep.QUOTA_CONSUMED_METHOD_NAME))
+        .thenReturn(returnedMethodConfiguration);
+    when(rawlsService.validateMethodConfig(
+            returnedMethodConfiguration,
+            PipelinesEnum.ARRAY_IMPUTATION.getValue(),
+            SubmitQuotaConsumedSubmissionStep.QUOTA_CONSUMED_METHOD_NAME,
+            TestUtils.TEST_PIPELINE_INPUTS_DEFINITION_LIST,
+            TestUtils.TEST_PIPELINE_OUTPUTS_DEFINITION_LIST,
+            TestUtils.TEST_WDL_METHOD_VERSION_1))
+        .thenReturn(true);
+    when(rawlsService.submitWorkflow(
+            eq("thisToken"),
+            submissionRequestCaptor.capture(),
+            eq(TestUtils.CONTROL_WORKSPACE_BILLING_PROJECT),
+            eq(TestUtils.CONTROL_WORKSPACE_NAME)))
+        .thenReturn(new SubmissionReport().submissionId(testJobId.toString()));
+
+    // do the step
+    SubmitQuotaConsumedSubmissionStep submitQuotaConsumedSubmissionStep =
+        new SubmitQuotaConsumedSubmissionStep(rawlsService, samService);
+    StepResult result = submitQuotaConsumedSubmissionStep.doStep(flightContext);
+
+    // extract the captured RunSetRequest and validate
+    SubmissionRequest submissionRequest = submissionRequestCaptor.getValue();
+    assertTrue(submissionRequest.isDeleteIntermediateOutputFiles());
+    assertFalse(submissionRequest.isUseReferenceDisks());
+    assertTrue(submissionRequest.isUseCallCache());
+    assertEquals(testJobId.toString(), submissionRequest.getEntityName());
+
+    // make sure the step was a success
+    assertEquals(StepStatus.STEP_RESULT_SUCCESS, result.getStepStatus());
+    assertEquals(
+        testJobId,
```

Comment:
this shouldn't be the testJobId, should be a different uuid (related to comment above)